common/web/types/test/helpers/reader-callback-test.ts

```diff
@@ -0,0 +1,109 @@
+import 'mocha';
+import {assert} from 'chai';
+import {loadLdmlKeyboardSchema, loadFile, makePathToFixture} from '../helpers/index.js';
+import LDMLKeyboardXMLSourceFileReader from '../../src/ldml-keyboard/ldml-keyboard-xml-reader.js';
+import { CompilerCallbacks, CompilerEvent } from '../../src/util/compiler-interfaces.js';
+import { LDMLKeyboardXMLSourceFile } from '../../src/ldml-keyboard/ldml-keyboard-xml.js';
+
+// TODO-LDML: this is largely a port from developer/src/kmc-keyboard/test/helpers/index.ts
```

Comment:
Stop me if you've seen this before…

these two are slightly different.  Might share some common code.

Reply:
Let's look at refactoring that as a possible future lowpri task

Reply:
it's already TODO-LDML.